### PR TITLE
Adds Demo page for school search

### PR DESCRIPTION
**Pull request by seb-siem-softwire**
Adds Demo page for school search

### CODE REVIEW

File: app/controllers/demo/school_search_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+module Demo
+  class SchoolSearchController < ApplicationController
+    def index
+
+    end
```
Comment (kicferk1) [inline app/controllers/demo/school_search_controller.rb:6]: We can make it shorter - maybe even remove it altogether. `def index; end`?

File: app/controllers/demo/school_search_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+module Demo
+  class SchoolSearchController < ApplicationController
+    def index
```
Comment (amazimbe) [inline app/controllers/demo/school_search_controller.rb:4]: How does rubocop feel about this? It usually prefers empty methods on a single line

File: app/models/school.rb [hunk 3]
```diff
@@ -18,8 +18,16 @@ class School < ApplicationRecord
   has_many :early_career_teacher_profiles
   has_many :early_career_teachers, through: :early_career_teacher_profiles, source: :user
 
+  scope :with_name, lambda { |search_key|
+    School.where("name ILIKE ?", "%#{search_key}%")
+  }
+
+  scope :with_urn, lambda { |search_key|
```
Comment (amazimbe) [inline app/models/school.rb:25]: 👍

File: app/controllers/demo/school_search_controller.rb (hunk 1 above)
Comment (amazimbe) [reply app/controllers/demo/school_search_controller.rb:6]: We could just remove it too.

Comment (seb-siem-softwire) [reply app/controllers/demo/school_search_controller.rb]: burned it with rubocop - burn baby burn

File: app/controllers/demo/school_search_controller.rb (hunk 2 above)
Comment (seb-siem-softwire) [reply app/controllers/demo/school_search_controller.rb]: burned it with rubocop

**Review by kicferk1 — APPROVED** [review]

**Comment by seb-siem-softwire** [comment]
Screenshot 2021-02-18 at 13 39 15

**Review by callumacrae — APPROVED** [review]

File: app/models/school.rb [hunk 4]
```diff
@@ -18,8 +18,16 @@ class School < ApplicationRecord
   has_many :early_career_teacher_profiles
   has_many :early_career_teachers, through: :early_career_teacher_profiles, source: :user
 
+  scope :with_name, lambda { |search_key|
```
Comment (ethanmills) [inline app/models/school.rb:21]: Is there a name for this that makes it more obvious that this matches part of the name? This sounds like `find_by_name` at the moment (to me at least)

Comment (ethanmills) [reply app/models/school.rb:21]: Maybe just `with_name_like`?

**Review by ethanmills — APPROVED** [review]
Looks good, just one comment about names

File: app/models/school.rb (hunk 4 above)
Comment (seb-siem-softwire) [reply app/models/school.rb]: Good point - changed

**Review by amazimbe — APPROVED** [review]

**Review by rupollbabu — APPROVED** [review]

**Pull request merged**